rust/otap-dataflow/crates/otap/src/otap_grpc/client_settings.rs

```diff
@@ -196,6 +223,123 @@ impl GrpcClientSettings {
             }
         }
     }
+
+    /// Returns the effective proxy configuration.
+    ///
+    /// If explicit proxy config is provided, it's merged with environment variables
+    /// (explicit values take precedence). If no explicit config is provided, reads
+    /// from environment variables only.
+    ///
+    /// # Performance
+    ///
+    /// This method reads environment variables (`HTTP_PROXY`, etc.) every time it is called.
+    /// It is intended to be called during client initialization (startup), not per-request.
+    #[must_use]
+    fn effective_proxy_config(&self) -> ProxyConfig {
+        match &self.proxy {
+            Some(config) => config.clone().merge_with_env(),
+            None => ProxyConfig::from_env(),
+        }
+    }
+
+    /// Logs an informational message if a proxy is configured for a plain text (http://) endpoint.
+    ///
+    /// This warns users that the configured proxy must support HTTP CONNECT for non-TLS targets,
+    /// which is a common source of connection failures with some proxy servers.
+    pub(crate) fn log_proxy_info(&self) {
+        let proxy = self.effective_proxy_config();
+        if proxy.has_proxy() && !self.grpc_endpoint.trim_start().starts_with("https://") {
+            let proxy_str = proxy.to_string();
+            otap_df_telemetry::otel_info!(
+                "Proxy.Configured",
+                endpoint = self.grpc_endpoint.as_str(),
+                proxy = proxy_str.as_str(),
+                message = "Proxy configured for http:// endpoint; using HTTP CONNECT tunneling. If your proxy does not support CONNECT for HTTP targets, consider using a transparent proxy or SOCKS proxy instead."
+            );
+        }
+    }
+
+    fn make_proxy_connector(
+        &self,
+        proxy: Arc<ProxyConfig>,
+    ) -> impl tower::Service<
+        http::Uri,
+        Response = TokioIo<tokio::net::TcpStream>,
+        Error = io::Error,
+        Future = impl Send + 'static,
+    > + Send
+    + Clone
+    + 'static {
+        // Capture settings at creation time.
+        // Note: If GrpcClientSettings are modified after this connector is created,
+        // those changes will NOT be reflected in the connector.
+        let tcp_nodelay = self.tcp_nodelay;
+        let tcp_keepalive = self.tcp_keepalive;
+        let tcp_keepalive_interval = self.tcp_keepalive_interval;
+        let tcp_keepalive_retries = self.tcp_keepalive_retries;
+
+        service_fn(move |uri: http::Uri| {
+            let proxy = Arc::clone(&proxy);
```

Comment:
The type system enforces Send + 'static bounds because tonic/tower are designed to be thread-safe generically. So we need to use `Arc` here.